sql/catalyst/src/main/scala/org/apache/spark/sql/types/StructType.scala

```diff
@@ -641,4 +641,30 @@ object StructType extends AbstractDataType {
     fields.foreach(s => map.put(s.name, s))
     map
   }
+
+  /**
+   * Returns a `StructType` that contains missing fields recursively from `source` to `target`.
+   * Note that this doesn't support looking into array type and map type recursively.
+   */
+  def findMissingFields(source: StructType, target: StructType, resolver: Resolver): StructType = {
```

Comment:
We need to define this method in the `StructType` side instead of a private method in `ResolveUnion`?

Reply:
I feel this is more general method related to `StructType`. So putting it in `StructType`.

Reply:
OK, it sounds good.